cmd/shell.go

```diff
@@ -255,10 +255,41 @@ type CAConfig struct {
 }
 
 // PAConfig specifies how a policy authority should connect to its
-// database, and what policies it should enforce.
+// database, what policies it should enforce, and what challenges
+// it should offer.
 type PAConfig struct {
 	DBConnect              string
 	EnforcePolicyWhitelist bool
+
+	EnableSimpleHTTP bool // TODO(#894) Remove this line
+	EnableDVSNI      bool // TODO(#894) Remove this line
+	EnableHTTP01     bool
+	EnableTLSSNI01   bool
+	EnableDNS01      bool
+}
+
+// SupportedChallenges returns the set of challenges supported by the
+// configuration, as a map[string]bool.
+func (pa PAConfig) SupportedChallenges() map[string]bool {
+	challenges := map[string]bool{}
```

Comment:
Since you turn the config values into a `map[string]bool` anyhow, why not just have a field in `PAConfig` that's a `map[string]bool`? You could have a list of known challenge type strings, iterate through the `map` kicking out any unknown values, and then choose a default map if `len(map) == 0`.

Alternately if you want a more strongly typed approach, you could define a `type EnabledChallenges struct {...}`, and pass that around instead of passing around a `map[string]bool`. I'd be happy with either approach.